packages/calypso-e2e/src/lib/flows/start-site-flow.ts

```diff
@@ -0,0 +1,101 @@
+import { Frame, Page } from 'playwright';
+import { BrowserHelper } from '../..';
+
+const selectors = {
+	// Generic
+	button: ( text: string ) => `button:text("${ text }")`,
+	backLink: 'a:text("Back")',
+
+	// Inputs
+	blogNameInput: 'input#siteTitle',
+	taglineInput: 'input#tagline',
+
+	// Themes
+	themePickerContainer: '.design-picker',
+	individualThemeContainer: ( name: string ) => `.design-button-container:has-text("${ name }")`,
+	previewThemeButtonDesktop: ( name: string ) => `button:has-text("Preview ${ name }")`,
+	previewThemeButtonMobile: ( name: string ) =>
+		`button.design-picker__design-option:has-text("${ name }")`,
+	themePreviewIframe: 'iframe[title=Preview]',
+};
+
+/**
+ * Class encapsulating the flow when starting a new start ('/start')
+ */
+export class StartSiteFlow {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the flow.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Given text, clicks on the first instance of the button with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 */
+	async clickButton( text: string ): Promise< void > {
+		await this.page.click( selectors.button( text ) );
+	}
+
+	/**
+	 * Enter blog name.
+	 *
+	 * @param name Name for the blog.
+	 */
+	async enterBlogName( name: string ): Promise< void > {
+		await this.page.fill( selectors.blogNameInput, name );
+	}
+
+	/**
+	 * Enter blog tagline.
+	 *
+	 * @param tagline Tagline for the blog.
+	 */
+	async enterTagline( tagline: string ): Promise< void > {
+		await this.page.fill( selectors.taglineInput, tagline );
+	}
+
+	/**
+	 * Validates we've landed on the design picker screen.
+	 */
+	async validateOnDesignPickerScreen(): Promise< void > {
+		await this.page.waitForSelector( selectors.themePickerContainer );
+	}
```

Comment:
>if we failed to land in the Theme Picker, then the subsequent attempt to pick a theme would fail.

Yes, I 100% agree with this! 👍 

In general, I've only been using `validate` methods if...
1. It's essential to testing to validate a piece of data on the screen...
2. That validation is not already done by some other natural action or step in the flow...
3. And it doesn't work to just return an exact value fetched from the DOM (often because React, and async state, etc.)

As a case example, later on in the `free` spec, we go to the settings page to validate that the site title and tagline we set way earlier in the flow actually made it saved to the site. We can't just grab the `value` directly from the `input` in the DOM because Playwright will outpace React state updating!

So, for this specific case, I only wanted to do this validation here to test a piece of navigation without completing the flow. When you go Hero Flow >> Start writing >> You get an option to pick a design or draft a first post. We ultimately will pick to draft a first post, completing the flow, but I wanted to test that the navigation was working correctly down this branch, which is why we select to pick a design but then navigate backwards. I wanted to make sure we _actually_ end up on the design picker, but we can't pick a design because that would complete the flow! Validating something on the webpage seemed more reliable than checking the URL, because the URL is a lie in React 😆 .

Does that make this function make more sense in this context?

I can take out the "Pick a design" case in this branch in the flow, but it seemed important to have included in some way.

Reply:
That's totally fine, I appreciate your clarification on this point. React is such a pain sometimes.

Reply:
> React is such a pain sometimes.
It is kind of a funny catch-22. The asynchronous, decoupled state updating of React is such a groundbreaking advancement in web development!

.... and it happens to make e2e testing brutal 🤣